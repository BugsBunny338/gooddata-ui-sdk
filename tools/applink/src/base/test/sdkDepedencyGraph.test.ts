// (C) 2020 GoodData Corporation
import flatten from "lodash/flatten";
import { determinePackageBuildOrder, findDependingPackages } from "../sdkDependencyGraph";
import { TestSdkDependencyGraph } from "./sdkDependencyGraph.fixture";

describe("dependency graph", () => {
    it("should create build order including all packages", () => {
        const buildOrder = determinePackageBuildOrder(TestSdkDependencyGraph);
        expect(flatten(buildOrder).length).toEqual(TestSdkDependencyGraph.nodes.length);
    });

    it("should create correct build order", () => {
        expect(determinePackageBuildOrder(TestSdkDependencyGraph)).toMatchSnapshot();
    });

    it("should create correct build order for prod deps only", () => {
        expect(determinePackageBuildOrder(TestSdkDependencyGraph, ["prod"])).toMatchSnapshot();
    });

    it("should find depending packages", () => {
        expect(findDependingPackages(TestSdkDependencyGraph, ["@gooddata/api-model-bear"])).toMatchSnapshot();
    });
});
