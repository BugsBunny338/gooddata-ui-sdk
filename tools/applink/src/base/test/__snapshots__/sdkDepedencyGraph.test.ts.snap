// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dependency graph should create correct build order 1`] = `
Array [
  Array [
    "@gooddata/sdk-skel-ts",
    "@gooddata/sdk-skel-tsx",
    "@gooddata/api-model-bear",
    "@gooddata/sdk-model",
    "@gooddata/util",
    "@gooddata/applink",
  ],
  Array [
    "@gooddata/sdk-backend-spi",
    "@gooddata/api-client-bear",
    "@gooddata/sdk-embedding",
  ],
  Array [
    "@gooddata/reference-workspace",
  ],
  Array [
    "@gooddata/api-client-tiger",
    "@gooddata/sdk-backend-base",
  ],
  Array [
    "@gooddata/sdk-backend-mockingbird",
    "@gooddata/sdk-backend-tiger",
    "@gooddata/catalog-export",
  ],
  Array [
    "@gooddata/sdk-ui",
    "@gooddata/sdk-backend-bear",
  ],
  Array [
    "@gooddata/mock-handling",
    "@gooddata/sdk-ui-filters",
    "@gooddata/sdk-ui-vis-commons",
    "@gooddata/sdk-ui-kit",
  ],
  Array [
    "@gooddata/sdk-ui-charts",
    "@gooddata/sdk-ui-pivot",
    "@gooddata/reference-workspace-mgmt",
    "@gooddata/live-examples-workspace",
    "@gooddata/experimental-workspace",
  ],
  Array [
    "@gooddata/sdk-ui-geo",
  ],
  Array [
    "@gooddata/sdk-ui-ext",
  ],
  Array [
    "@gooddata/sdk-ui-tests",
    "@gooddata/sdk-ui-all",
  ],
]
`;
